opengever/api/listing.py

```
@@ -200,7 +203,80 @@ def extract_sort(self, params, query):
                 sort += ' desc'
             else:
                 sort += ' asc'
-        return sort
+
+        return self.extend_with_sort_first(sort, params)
+
+    def extend_with_sort_first(self, sort, params):
+        """The sort_first sorts docs in two steps. First, it will sort
+        all items having one of the provided values of a defined field first.
+        All other docs will be put into the second group. Each group will then
+        be sorted by the `sort_order`.
+
+        Having a result of the following docs ordered by sortable_title:
+
+        { 'portal_type': 'type1', title: 'A' }
+        { 'portal_type': 'type2', title: 'B' }
+        { 'portal_type': 'type2', title: 'C' }
+        { 'portal_type': 'type3', title: 'D' }
+        { 'portal_type': 'type1', title: 'E' }
+        { 'portal_type': 'type4', title: 'F' }
+
+        Can be grouped i.e. by type:
+
+        { 'portal_type': ['type1', type3'] }
+
+        which will result in the following sort order:
+
+        { 'portal_type': 'type1', title: 'A' } <= Group 1
+        { 'portal_type': 'type3', title: 'D' }
+        { 'portal_type': 'type1', title: 'E' }
+        { 'portal_type': 'type2', title: 'B' } <= Group 2
+        { 'portal_type': 'type2', title: 'C' }
+        { 'portal_type': 'type4', title: 'F' }
+        """
+        sort_first = params.get('sort_first', {})
+        if not isinstance(sort_first, record):
+            sort_first = {}
```

I'd rather raise a BadRequest here than silently dropping the user input